### PR TITLE
libftdi.py: Improve error handling

### LINKED ISSUE #21: Error `name 'ftdi' is not defined` when native library isn't installed
The issue is in `pyft232/ft232/libftdi.py`, in lines 30/31 any exception loading the library is ignored, but then in line 36 it tries to use `ftdi` despite the fact that the object is not created if there is an error.  Instead, lines 30/31 should throw a meaningful exception that tells the user that the native library was not found.

### PR DESCRIPTION
**Pull request by qwertychouskie**
Closes https://github.com/lsgunth/pyft232/issues/21

If you get the chance, publishing a new release to PyPI after this is merged would be a huge help, as the now-merged #22 is necessary for the macOS release of our software, and this PR also improves things a bit.

### CODE REVIEW

File: ft232/libftdi.py [hunk 1]
```diff
@@ -23,13 +23,22 @@
 from serial import (FIVEBITS, SIXBITS, SEVENBITS, EIGHTBITS, PARITY_NONE,
                     PARITY_EVEN, PARITY_ODD, STOPBITS_ONE, STOPBITS_TWO)
 
-for lib in ("libftdi.so", "libftdi.so.1", "libftdi.dylib", "libftdi.dylib.1", "libftdi1.dylib"):
+libftdi_filenames = ["libftdi.so", "libftdi.so.1", "libftdi.dylib", "libftdi.dylib.1", "libftdi1.dylib"]
+
+for lib in libftdi_filenames:
     try:
         ftdi = c.CDLL(lib)
+        print("Loaded libftdi: ", lib)
```
Comment (lsgunth) [inline ft232/libftdi.py]: I don't think a library should be printing things when it's loaded.

File: ft232/libftdi.py [hunk 2]
```diff
@@ -23,13 +23,22 @@
...
         break
     except OSError:
         continue
 
+try:
+    ftdi
```
Comment (lsgunth) [inline ft232/libftdi.py]: This is a bit ugly. Better to just use an else on the for loop:

```python
for lib in libftdi_filenames:
    try:
        ftdi = c.CDLL(lib)
        break
    except OSError:
        continue
else:
    raise ...
```

File: ft232/libftdi.py [hunk 3]
```diff
@@ -23,13 +23,22 @@
...
+except:
+    print("Error: Unable to load libftdi!  Please make sure it is available as at least one of the following filenames:\n" + "\n".join(libftdi_filenames))
+    raise Exception("Unable to load libftdi!  Please make sure it is available as at least one of the following filenames:\n" + "\n".join(libftdi_filenames))
```
Comment (lsgunth) [inline ft232/libftdi.py]: It's poor form to raise a bare Exception seeing it can't be caught by anything. I'd suggest raising a FileNotFoundError instead.

**Review by lsgunth — COMMENTED** [review]
Thanks! I have just a couple notes to improve your patch.

**Comment by qwertychouskie** [comment]
Alright, changes made and tested, it should be good to go for merging.

File: ft232/libftdi.py [hunk 4]
```diff
@@ -23,12 +23,17 @@
...
+libftdi_filenames = ["libftdi.so", "libftdi.so.1", "libftdi.dylib", "libftdi.dylib.1", "libftdi1.dylib"]
+
+for lib in libftdi_filenames:
     try:
         ftdi = c.CDLL(lib)
         break
     except OSError:
         continue
+else:
+    print("Error: Unable to load libftdi!  Please make sure it is available as at least one of the following filenames:\n" + "\n".join(libftdi_filenames))
```
Comment (lsgunth) [inline ft232/libftdi.py]: Thanks! Can you just please drop this print as well? The exception should be more than enough and if a user wants to catch the exception and handle this in another way they probably don't want the extra noise.

Comment (qwertychouskie) [reply ft232/libftdi.py]: Alright, fixed.

**Review by lsgunth — APPROVED** [review]
Thanks!

**Pull request merged**